backend/cmd/api/main.go

```diff
@@ -55,6 +61,48 @@ func playgroundHandler() gin.HandlerFunc {
 	}
 }
 
+func previewCollectionHandler(loaders *common.BatchLoaders) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		//TODO: authentication & authorization
+		collectionId := c.Param("id")
+		intID, err := strconv.ParseInt(collectionId, 10, 32)
+		if err != nil {
+			c.AbortWithStatus(http.StatusBadRequest)
+			return
+		}
+		loaders.CollectionLoader.Clear(c, int(intID))
+		loaders.CollectionItemLoader.Clear(c, int(intID))
+		loaders.CollectionItemIdsLoader.Clear(c, int(intID))
+		items, err := collection.GetCollectionItems(c, loaders, int(intID))
+		if err != nil {
+			c.AbortWithStatus(http.StatusInternalServerError)
+			return
+		}
+		var result []map[string]any
+
+		for index, item := range items {
+			if index >= 20 {
+				break
+			}
+			if err != nil {
+				c.AbortWithStatus(http.StatusInternalServerError)
+				return
+			}
+
+			t := reflect.TypeOf(item).Elem().Name()
+
+			stringed, _ := json.Marshal(item)
+			var mapped map[string]any
+			_ = json.Unmarshal(stringed, &mapped)
+			mapped["type"] = strings.ToLower(t)
+
+			result = append(result, mapped)
+		}
+
+		c.JSON(200, result)
+	}
+}
+
```

Comment:
Why don't we allow just getting the collection by id via GQL? Can be a new endpoint but I'm kind of reluctant to start adding a REST based api here now.

Reply:
Because of cache. This endpoint is for internal use.

Could move this to another API for internal use?
Just need to use existing go code.

Reply:
I don't think we should expose internal endpoints through the same gql schema, but making an admin api could solve our "issues"